src/components/Pages/RegisterPage.tsx

```diff
@@ -51,17 +52,89 @@ const Button = styled.button`
 `;
 
 const RegisterPage = () => {
+	const [values, setValues] = useState({
+		firstname: '',
+		lastname: '',
+		username: '',
+		email: '',
+		password: '',
+		confirmPassword: '',
+	});
+
+	const inputs = [
+		{
+			id: 1,
+			name: 'firstname',
+			type: 'text',
+			placeholder: 'First Name',
+		},
+		{
+			id: 2,
+			name: 'lastname',
+			type: 'text',
+			placeholder: 'Last Name',
+		},
+		{
+			id: 3,
+			name: 'username',
+			type: 'text',
+			placeholder: 'username',
+			errorMessage:
+				'Username should be 3-16 characters and shouldnt include any special character!',
+			pattern: '^[A-Za-z0-9]{3,16}$',
+			required: true,
+		},
+		{
+			id: 4,
+			name: 'email',
+			type: 'email',
+			placeholder: 'Email',
+			errorMessage: 'It should be valid email address!',
+			required: true,
+		},
+		{
+			id: 5,
+			name: 'password',
+			type: 'password',
+			placeholder: 'Password',
+			errorMessage:
+				'Password should be 8-20 characters and include at least 1 letter, 1 number, and 1 special character!',
+			pattern: `^(?=.*[0-9])(?=.*[a-zA-Z])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,20}$`,
+			required: true,
+		},
+		{
+			id: 6,
+			name: 'confirmPassword',
+			type: 'password',
+			placeholder: 'Confirm Password',
+			errorMessage: 'Password dont match!',
+			pattern: values.password,
+			required: true,
+		},
+	];
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		console.log(values);
+	};
+
+	const onChange = (event) => {
+		setValues({ ...values, [event.target.name]: event.target.value });
```

Comment:
use setState with callback to reference values - it's much safer as then you're sure values are up to date even if they were changed by different code:
https://reactjs.org/docs/hooks-reference.html#usestate:~:text=setState(prevState%20%3D%3E%20%7B%0A%20%20//%20Object.assign%20would%20also%20work%0A%20%20return%20%7B...prevState%2C%20...updatedValues%7D%3B%0A%7D)%3B